### ISSUE TITLE
ENH: add `BaseEpochs.to_torch_dataset()` for native `PyTorch` integration

### ISSUE THREAD

**Issue by aman-coder03** [mne-tools/mne-python#13730]
### Describe the new feature or enhancement

right now there is no built-in way to go from `mne.Epochs` to a `torch.utils.data.Dataset`. Every user doing deep learning with EEG/MEG ends up writing the same glue code manually...
```python
data = epochs.get_data(picks='eeg')
labels = epochs.events[:, 2]  # integers, not condition names
dataset = torch.utils.data.TensorDataset(
    torch.FloatTensor(data), torch.LongTensor(labels)
)
```
this loses condition name labels, bypasses MNE's existing `picks` and `tmin`/`tmax` handling, and gets rewritten from scratch by everyone. `to_torch_dataset()` method would fix this the same way `to_data_frame()` did for pandas

### Describe your proposed implementation

add `to_torch_dataset()` to `BaseEpochs` in `mne/epochs.py`, right after `to_data_frame()`,it follows the exact same 3 step pattern...
```python
def to_torch_dataset(self, picks=None, label_key='condition',
                     tmin=None, tmax=None, *, verbose=None):
    torch = _check_torch_installed()  # same guard as _check_pandas_installed()
    data = self.get_data(picks=picks, tmin=tmin, tmax=tmax, copy=True)
    rev_event_id = {v: k for k, v in self.event_id.items()}
    labels = [rev_event_id[k] for k in self.events[:, 2]]
    return _MNEEpochsDataset(torch.as_tensor(data, dtype=torch.float32), labels)
```

`_MNEEpochsDataset` is a small private `Dataset` subclass with just `__len__` and `__getitem__`

### Describe possible alternatives

1) use `braindecode/torcheeg` they solve this but are heavy dependencies for users who just want a DataLoader, not a full DL framework
2) standalone function instead of a method does not fit mne's existing style where all export methods live on the object

### Additional context

`torch` would stay fully optional, guarded the same way `pandas` is today...
no impact on users who don't have it installed... happy to open a PR

**Comment by agramfort** [comment]
-1 to have a torch deps in mne.

also this likely does not work well for any scalable system where data does
not fit in memory.

I would go the braindecode route and discuss there how to make it more
lightweight if needs be

**Comment by aman-coder03** [comment]
thanks @agramfort i'll close this and open an issue in `braindecode` to add mapping, preload and picks support to `create_from_mne_epochs()` to make it more lightweight, will discuss there
